components/admin_console/text_setting.tsx

```diff
@@ -1,30 +1,36 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 
 import TextSetting from 'components/widgets/settings/text_setting';
 
 import SetByEnv from './set_by_env';
 
-const AdminTextSetting = (props) => {
-    const {setByEnv, disabled, ...sharedProps} = props;
+interface Props {
```

Comment:
This type definition is not what was expected, in this case it was expected to receive the same props that were in `TextSetting` from widgets (you can get it exporting it in the `components/widgets/settings/text_setting.tsx` file, and importing it here using something like `import TextSetting. {Props as TextSettingProps} from 'components/widgets/settings/text_setting';`). To apply the type with extra fields you should use something like `type Props = TextSettingProps & {setByEnv: boolean}`.

Reply:
Ah. This is a good approach